deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_message.erl

```diff
@@ -282,3 +302,52 @@ map_add(_T, _Key, _Type, undefined, Acc) ->
     Acc;
 map_add(KeyType, Key, Type, Value, Acc) ->
     [{wrap(KeyType, Key), wrap(Type, Value)} | Acc].
+
+amqp10_app_props_to_amqp091_headers(CurrentHeaders, AppPropsBin) -> 
+    case amqp10_framing:decode_bin(AppPropsBin) of 
+        [#'v1_0.application_properties'{ content = AppProps}] when is_list(AppProps) -> 
+            Hs = case CurrentHeaders of 
+                undefined -> [];
+                Headers -> Headers
+            end,
+            lists:foldl(fun(Prop, Acc) -> 
+                case Prop of 
+                    {{utf8, Key}, {ValueType, Value}} -> 
+                        case type10_to_type091(Key, ValueType, Value) of 
+                            undefined -> Acc;
+                            Typed -> [Typed |Acc]
+                        end;
+                    _ -> Acc
+                end
+            end, Hs, AppProps);
+        _ -> CurrentHeaders
+    end.
+type10_to_type091(Key, Type, Value) -> 
+    try
+        rabbit_msg_record:to_091(Key, {Type, Value})
```

Comment:
Do we know which types will cause a function_clause?  perhaps we should just have a fall back clause in the rabbit_msg_record:to_091 function?

Reply:
I am not sure which type causes an error, I just defensively added the try/catch as I did not verify that all types are handled according to the specs. I think it would be better to return `undefined`.

Reply:
In this case though the function would lose it's nice property that the return value is a valid header value which can be immediately encoded as binary. Now we would need to filter the results of all these calls. 

If you think all conversions cases are handled  I can just remove the try catch, but I like to be on the safe side. 😄